src/import-export/csv-imp/gnc-tx-import.cpp

```diff
@@ -0,0 +1,960 @@
+                                 "(?<DAY>[0-9]+)"
+                             "|"                                     // or CCYYMMDD
+                                 "(?<YEAR>[0-9]{4})"
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<DAY>[0-9]{2})"
+                             ")",
+
+                             "(?:"                                   // either d-m-y
+                                 "(?<DAY>[0-9]+)[-/.' ]+"
+                                 "(?<MONTH>[0-9]+)[-/.' ]+"
+                                 "(?<YEAR>[0-9]+)"
+                             "|"                                     // or DDMMCCYY
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<YEAR>[0-9]{4})"
+                             ")",
+
+                             "(?:"                                   // either m-d-y
+                                 "(?<MONTH>[0-9]+)[-/.' ]+"
+                                 "(?<DAY>[0-9]+)[-/.' ]+"
+                                 "(?<YEAR>[0-9]+)"
+                             "|"                                     // or MMDDCCYY
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<YEAR>[0-9]{4})"
+                             ")",
+
+                             "(?:"                                   // either d-m(-y)
+                                 "(?<DAY>[0-9]+)[-/.' ]+"
+                                 "(?<MONTH>[0-9]+)(?:[-/.' ]+"
+                                 "(?<YEAR>[0-9]+))?"
+                             "|"                                     // or DDMM(CCYY)
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<YEAR>[0-9]+)?"
+                             ")",
+
+                             "(?:"                                   // either m-d(-y)
+                                 "(?<MONTH>[0-9]+)[-/.' ]+"
+                                 "(?<DAY>[0-9]+)(?:[-/.' ]+"
+                                 "(?<YEAR>[0-9]+))?"
+                             "|"                                     // or MMDD(CCYY)
+                                 "(?<MONTH>[0-9]{2})"
+                                 "(?<DAY>[0-9]{2})"
+                                 "(?<YEAR>[0-9]+)?"
+                             ")",
+};
+//const int num_currency_formats = 3;
+//const gchar* currency_format_user[] = {N_("Locale"),
+//                                       N_("Period: 123,456.78"),
+//                                       N_("Comma: 123.456,78")
+//                                      };
+//
+/* This map contains a set of strings representing the different column types. */
+std::map<GncTransPropType, const char*> gnc_csv_col_type_strs = {
```

Comment:
The casting error is (IIUC) because you used enum class instead of just enum; if at gnc-tx-import.cpp:50 you say ```enum GncTransPropType :  std::vector<std::string>::size_type```
it should implicitly convert. Was there a reason for enum class?

As for the iteration, hmm. I didn't see that; the file is reduced to a one-liner in Github ("change set too big, please look at it offline") and I missed the whole file. The S/O article objects to iterating over enums in the general case because they might be discontinuous. This one isn't so that's not really a concern, but you'd still have to switch to an old-style for loop (```for(auto type = GncTransPropType::NONE; type < GncTransPropType::Sentinal; ++type)```) and that's a bit ugly. You could make a ```std::vector<std::pair<GncTransPropType, std::string>>``` and iterate over that, but changing now might be premature optimization.